src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeUtils.cs

```diff
@@ -730,5 +730,12 @@ internal static bool IsFloatingPoint(Type type)
             }
         }
 #endif 
+
+        public static bool IsVector(this Type type)
+        {
+            // Unfortunately, the IsSzArray property of System.Type is inaccessible to us,
+            // so we use a little equality comparison trick instead:
+            return type == type.GetElementType().MakeArrayType();
```

Comment:
If the object references are equal, the types are the same.  But is it guaranteed that if they're not equal they're not the same type?  I know the runtime caches Type objects, but I'm not sure what the caching policy is, whether it's based on weak references (in which case this should be fine) or something else (in which case maybe there's a problem?)